### PR TITLE
Bookmark feature

### PR DESCRIPTION
**Pull request by MeenaFoerderer**

### CODE REVIEW

File: pages/[date]/[room]/index.js [hunk 1]
```diff
@@ -55,38 +65,94 @@ function Room({ conferenceDays, conferenceRooms, talks }) {
...
+`;
+
+const Button = styled.button`
+    background-color: transparent;
+    margin-top: -5px;
+    margin-right: -10px;
+    border: none;
+    
+    &:hover {
+    background-color: transparent;
```
Comment (JessicaLoers) [inline pages/[date]/[room]/index.js:153]:
```suggestion
    &:hover {
    background-color: transparent;
    }
```

File: pages/bookmarks/index.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+          ...talk,
+          isBookmarked: !talk.isBookmarked,
+        };
+      } else {
+        return talk;
+      }
+    });
+
+    setTalks(newTalksArray);
+  }
```
Comment (JessicaLoers) [inline pages/bookmarks/index.js:18]:
```suggestion

```

File: pages/bookmarks/index.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+      <ul>
+        {talks?.map((talk) => {
+          if (talk.isBookmarked === true) {
+            return <li toggleFavorite={toggleFavorite}>{talk.title}</li>;
+          } else {
+            return null;
+          }
+        })}
+      </ul>
```
Comment (JessicaLoers) [inline pages/bookmarks/index.js:32]:
```suggestion
      <ul>
      <ul>{talks.map((talk) => talk.isBookmarked && <li key={talk.id}>{talk.title}</li>)}</ul>
      </ul>
```

File: pages/_app.js [hunk 4]
```diff
@@ -1,8 +1,11 @@
 import GlobalStyles from "../components/GlobalStyles";
 import { data } from "../helpers/data";
 import { dateFromNormalizedString } from "../helpers/normalize";
+import { useLocalStorage } from "../helpers/hooks";
 
 function MyApp({ Component, pageProps }) {
+  const [talks, setTalks] = useLocalStorage("talks", data);
+
   const conferenceDays = Array.from(new Set(data.map((talk) => talk.date)))
```
Comment (JessicaLoers) [inline pages/_app.js:9]:
```suggestion
  const conferenceDays = Array.from(new Set(talks.map((talk) => talk.date)))
```

File: pages/_app.js [hunk 5]
```diff
@@ -11,14 +14,22 @@ function MyApp({ Component, pageProps }) {
     new Set(data.map((talk) => talk.room))
```
Comment (JessicaLoers) [inline pages/_app.js:14]:
```suggestion
    new Set(talks.map((talk) => talk.room))
```

File: pages/[date]/[room]/talks/[id].js [hunk 6]
```diff
@@ -17,26 +22,50 @@ function TalkDetails() {
     day,
     date: talkDate,
     time,
-  } = data.find((talk) => talk.id === id);
+    isBookmarked,
+  } = talks.find((talk) => talk.id === id);
```
Comment (JessicaLoers) [inline pages/[date]/[room]/talks/[id].js:26]:
```suggestion
 const {
    title,
    authors,
    abstract,
    session,
    day,
    date: talkDate,
    time,
    id: talkId,
    isBookmarked,
  } = talks.find((talk) => talk.id === id);
```

File: pages/[date]/[room]/talks/[id].js [hunk 7]
```diff
@@ -17,26 +22,50 @@ function TalkDetails() {
...
+            <CloseIcon />
+          </Link>
+          <Button
+            type={"button"}
+            onClick={() => {
+              onBookmarkToggle(talks.id);
+            }}
+          >
+            {talks.isBookmarked ? <BookmarkActive /> : <BookmarkInactive />}
+          </Button>
```
Comment (JessicaLoers) [inline pages/[date]/[room]/talks/[id].js:50]:
```suggestion
          <Button
            type={"button"}
            onClick={() => {
              onBookmarkToggle(talkId)
            }}
          >
            {isBookmarked ? <BookmarkActive /> : <BookmarkInactive />}
          </Button>
```

File: pages/[date]/[room]/talks/[id].js [hunk 8]
```diff
@@ -46,37 +75,76 @@ const StyledMain = styled.div`
...
-  border-radius: 5px;
-  border: none;
-  background-color: #eaeaea;
-  display: flex;
-  justify-content: center;
-  align-items: center;
+const IconWrapper = styled.div`
+display: flex:
+justify-content: space-between;
+`;
```
Comment (JessicaLoers) [inline pages/[date]/[room]/talks/[id].js:81]:
```suggestion
const IconWrapper = styled.div`
display: flex;
justify-content: space-between;
`;
```

File: pages/[date]/[room]/talks/[id].js [hunk 9]
```diff
@@ -46,37 +75,76 @@ const StyledMain = styled.div`
...
 `;
 
+const Button = styled.button`
+    background-color: transparent;
+    margin-top: -5px;
+    margin-right: -10px;
+    border: none;
+    
+    &:hover {
+    background-color: transparent;
```
Comment (JessicaLoers) [inline pages/[date]/[room]/talks/[id].js:147]:
```suggestion
    &:hover {
    background-color: transparent;
    }
```

**Review by JessicaLoers — COMMENTED** [review]
Please have a look at my suggestions 🙂

File: pages/[date]/[room]/talks/[id].js (hunk 6 above)
Comment (MeenaFoerderer) [reply pages/[date]/[room]/talks/[id].js:27]: This was the fundamental change, thanks!

File: pages/[date]/[room]/talks/[id].js (hunk 8 above)
Comment (MeenaFoerderer) [reply pages/[date]/[room]/talks/[id].js]: Haven't seen this tiny mistake but was really confused why it didn't work, thanks!

File: pages/[date]/[room]/index.js [hunk 10]
```diff
@@ -128,29 +168,53 @@ const TalkList = styled.ul`
...
-  font-size: 1rem;
+const HomeIcon = styled(BsHouseDoor)`
+  font-size: 2.5em;
+  color: #f5f5f5;
 `;
 
+const BookmarkIcon = styled(BsJournalBookmarkFill)`
+  font-size: 2.5em;
+  color: #f5f5f5;
+`;
```
Comment (ahohnsen) [inline pages/[date]/[room]/index.js:219]: see my previous comment

File: pages/[date]/[room]/talks/[id].js [hunk 11]
```diff
@@ -17,66 +27,148 @@ function TalkDetails() {
...
+              <CloseIcon />
+            </BackButton>
+            <ToggleButton
+              type={"button"}
+              onClick={() => {
+                onBookmarkToggle(talkId);
+              }}
+            >
+              {isBookmarked ? <BookmarkActive /> : <BookmarkInactive />}
+            </ToggleButton>
```
Comment (ahohnsen) [inline pages/[date]/[room]/talks/[id].js:57]: Please add an `aria-label` here as you have done for your `BackButton`. I suggest to check all places, where you are rendering an SVG and to add an `aria-label`, if necessary.

**Review by ahohnsen — CHANGES_REQUESTED** [review]
Hey Meena,

Nicely done ✨ I added some comments. 

Please check, that you have a `main` element on every page. Additionally, all links and buttons, which only include an svg and no text content, should have an `aria-label`. 🙂

**Review by ahohnsen — APPROVED** [review]
Hey Meena,

Please add a `main `element to all your pages and then you are good to go 🙂

**Comment by MeenaFoerderer** [comment]
> Please add a `main `element to all your pages and then you are good to go 🙂

Oh, I completely overlooked that. Thanks for approving!

**Pull request merged**